Refactor `*person*` into `*module*`

Creastery opened the pull request:
Disclaimer: This is going to be an extremely huge Pull Request, so please help to find out if I missed out or messed up anything.

If there is any issues, please let me know at the soonest so that I can fix it :)
Thanks in advance for helping to review this huge PR!

Proposed commit message: 
```
We are going to morph the `persons`-oriented Address Book application to a
`modules`-oriented Degree Planner application.

Before we can start tweaking existing features and implementing new
features to suit our degree planner application, we need to refactor the
codebase so that we are working with `modules` instead of `persons`.

As the codebase is quite huge, we carry out the refactoring in multiple
stages incrementally to ensure the correctness of the refactoring done.

In the first phase of refactoring, we retain all existing
functionalities, but avoid updating any variables (such as strings or
variable names referenced by Jackson JSON processor for serialization)
or filenames that are referenced by unit tests so as to allow all unit
tests to continue to pass.

In the second phase of refactoring, we update all variables and
filenames, and amend unit tests accordingly so that all unit tests
continue to pass.

In the third phase of refactoring, we update the documentations
(e.g. User Guide and Developer Guide).

Let's refactor the entire codebase in the three refactoring phases above
incrementally such that we eventually refactor
* filenames that are `*person*` to `*module*`
* class names that are `*person*` to `*module*`
* class attributes are `*person*` to `*module*`
* class methods are `*person*` to `*module*`
* all variables are `*person*` to `*module*`
* `*person*` to `*module*` in comments
* all documentations to replace `*person*` with `*module*`
* all documentations to include updated `*module*`-related diagrams

In addition, we ensure that unit tests continue to pass so that each 
refactoring step taken is easily verifiable.

First phase of refactoring:

  [ 1/68] Model: refactor package `person` to `module`
  [ 2/68] Model: refactor class `Person` to `Module`
  [ 3/68] DuplicatePersonException class: refactor to `DuplicateModuleException`
  [ 4/68] PersonNotFoundException class: refactor to `ModuleNotFoundException`
  [ 5/68] Model: refactor class `UniquePersonList` to `UniqueModuleList`
  [ 6/68] Model: refactor `Model` interface
  [ 7/68] ReadOnlyAddressBook: refactor `getPersonList()` to `getModuleList()`
  [ 8/68] Logic: refactor `Logic` interface method names
  [ 9/68] commons/core: refactor `Messages` class attribute names
  [10/68] Logic/commands: refactor `AddCommand` class attributes names
  [11/68] Logic/commands: refactor `DeleteCommand` class attribute names
  [12/68] Logic/commands: refactor `EditCommand#EditPersonDescriptor` class
  [13/68] Logic/commands: refactor `EditCommand` class
  [14/68] Logic/commands: refactor `FindCommand` class comments
  [15/68] Logic/commands: refactor `ListCommand` class comments
  [16/68] Logic/commands: refactor `SelectCommand` class attributes
  [17/68] Model/util: refactor `SampleDataUtil#getSamplePersons()`
  [18/68] Model: refactor `AddressBook` class methods
  [19/68] Model: refactor `ModelManager` class
  [20/68] Storage: refactor class `JsonAdaptedPerson` to `JsonAdaptedModule`
  [21/68] Storage: refactor `JsonSerializableAddressBook` class attributes
  [22/68] UI: refactor `BrowserPanel` class
  [23/68] UI: refactor class `PersonCard` to `ModuleCard`
  [24/68] UI: refactor class `PersonListPanel` to `ModuleListPanel`
  [25/68] guitests: refactor class `PersonCardHandle` to `ModuleCardHandle`
  [26/68] guitests: refactor `PersonListPanelHandle` to `ModuleListPanelHandle`
  [27/68] test/commons: refactor `IndexTest` class variable names
  [28/68] test/testutil: refactor `AddressBookBuilder#withPerson(...)`
  [29/68] test/testutil: refactor class `PersonBuilder` to `ModuleBuilder`
  [30/68] test/testutil: refactor class `PersonUtil` to `ModuleUtil`
  [31/68] test/testutil: refactor `TestUtil#getPerson(...)` to `getModule(...)`
  [32/68] test/testutil: refactor `TypicalIndexes` class attributes
  [33/68] test/testutil: refactor class `TypicalPersons` to `TypicalModules`
  [34/68] test/testutil: refactor `EditPersonDescriptorBuilder` class
  [35/68] test/commands: refactor `AddCommand` and `AddCommandIntegrationTest`
  [36/68] test/commands: refactor `CommandTestUtil` class methods
  [37/68] test/commands: refactor `DeleteCommandTest` class methods
  [38/68] test/commands: refactor `EditCommandTest` class
  [39/68] test/commands: refactor `FindCommandTest` class methods
  [40/68] test/commands: refactor `SelectCommandTest` class variable names
  [41/68] test/logic: refactor `LogicManagerTest` class methods
  [42/68] test/model: refactor class `PersonTest` to `ModuleTest`
  [43/68] test/model: refactor `LogicManagerListTest` class methods names
  [44/68] test/model: refactor `AddressBookTest` class method names
  [45/68] test/model: refactor `ModelManagerTest` class method names
  [46/68] test/storage: refactor `JsonAdaptedModuleTest` class
  [47/68] test/storage: refactor `JsonAddressBookStorageTest` class method names
  [48/68] test/storage: refactor `JsonAddressBookStorageTest` class
  [49/68] test/ui: refactor `GuiTestAssert` class
  [50/68] test/ui: refactor `BrowserPanelTest` class attributes
  [51/68] test/ui: refactor `ModuleListPanelTest` class
  [52/68] systemtests: refactor `AddressBookSystemTest` class
  [53/68] systemtests: refactor `DeleteCommandSystemTest` class
  [54/68] systemtests: refactor `FindCommandSystemTest` class comments
  [55/68] systemtests: refactor `ModelHelper` class
  [56/68] systemtests: refactor `SelectCommandSystemTest` class variable names
  
Second refactoring phase:

  [57/68] UI: refactor `ModuleListPanel#moduleListView` class attribute name
  [58/68] UI: refactor `MainWindow#personListPanelPlaceholder` class attribute
  [59/68] UI: refactor `PersonListCard.fxml` to `ModuleListCard.fxml`
  [60/68] UI: refactor `PersonListPanel.fxml` to `ModuleListPanel.fxml`
  [61/68] Storage: refactor `JsonSerializableAddressBook#persons` to `modules`
  [62/68] test/data: refactor `JsonAddressBookStorageTest` JSON files
  [63/68] test/data: refactor `JsonSerializableAddressBookTest` JSON files
  [64/68] Replace `*person*` with `*module*` in all string literals

Third refactoring phase:

  [65/68] DeveloperGuide: rename filepaths of sequence diagrams
  [66/68] docs: update diagrams and images
  [67/68] docs: replace `*person*` with `*module*` in User and Developer guides
  [68/68] docs/DeveloperGuide: replace `*person*` with `*module*`
```

Creastery committed:
Model: refactor package `person` to `module`

We are going to morph a `persons`-oriented Address Book application to a
`modules`-oriented Degree Planner application.

Before we can start tweaking existing features and implementing new
features to suit our degree planner application, we need to refactor the
codebase so that we are working with `modules` instead of `persons`.

As the codebase is quite huge, we carry out the refactoring in multiple
stages incrementally to ensure the correctness of the refactoring done.

In the first phase of refactoring, we retain all existing
functionalities, but avoid updating any variables (such as strings or
variable names referenced by Jackson JSON processor for serialization)
or filenames that are referenced by unit tests so as to allow all unit
tests to continue to pass.

In the second phase of refactoring, we update all variables and
filenames, and amend unit tests accordingly so that all unit tests
continue to pass.

In the third phase of refactoring, we update the documentations
(e.g. User Guide and Developer Guide).

As an initial step of the first refactoring phase, let's rename the
model package `person` to `module`.

Creastery committed:
Model: refactor class `Person` to `Module`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class `Person` to
`Module` and:
* refactor class method names containing `*person*` to `*module*`
* refactor class attributes names containing `*person*` to `*module*`
* refactor variable names containing `*person*` to `*module*`
* replace `*person*` with `*module*` in comments
* cascade the changes to all affected classes

Creastery committed:
DuplicatePersonException class: refactor to `DuplicateModuleException`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class
`DuplicatePersonException` to `DuplicateModuleException` and:
* refactor class method names containing `*person*` to `*module*`
* replace `*person*` with `*module*` in comments

Creastery committed:
PersonNotFoundException class: refactor to `ModuleNotFoundException`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class
`PersonNotFoundException` to `ModuleNotFoundException` and cascade the
changes to all affected classes.

Creastery committed:
Model: refactor class `UniquePersonList` to `UniqueModuleList`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class
`UniquePersonList` to `UniqueModuleList` and:
* refactor class method names containing `*person*` to `*module*`
* refactor class attributes names containing `*person*` to `*module*`
* refactor variable names containing `*person*` to `*module*`
* refactor respective unit testing class name
* replace `*person*` with `*module*` in comments
* cascade the changes to all affected classes

Creastery committed:
Model: refactor `Model` interface

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `Model` interface by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring class attributes names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
ReadOnlyAddressBook: refactor `getPersonList()` to `getModuleList()`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`ReadOnlyAddressBook#getPersonList()` to `getModuleList()` and:
* replace `*person*` with `*module*` in comments
* cascade the changes to all affected classes

Creastery committed:
Logic: refactor `Logic` interface method names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `Logic` interface by:
* refactoring interface method names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
commons/core: refactor `Messages` class attribute names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `Messages` class by:
* refactoring class attribute names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
Logic/commands: refactor `AddCommand` class attributes names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `AddCommand` class
by:
* refactoring class attribute names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
Logic/commands: refactor `DeleteCommand` class attribute names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `DeleteCommand` class
by:
* refactoring class attribute names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
Logic/commands: refactor `EditCommand#EditPersonDescriptor` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`EditCommand#EditPersonDescriptor` nested class to
`EditCommand#EditModuleDescriptor` and:
* refactor class method names containing `*person*` to `*module*`
* refactor class attributes names containing `*person*` to `*module*`
* refactor variable names containing `*person*` to `*module*`
* cascade the changes to all affected classes

Creastery committed:
Logic/commands: refactor `EditCommand` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `EditCommand` class
by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring class attributes names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
Logic/commands: refactor `FindCommand` class comments

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `FindCommand` class
by replacing `*person*` with `*module*` in comments.

Creastery committed:
Logic/commands: refactor `ListCommand` class comments

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `ListCommand` class
by replacing `*person*` with `*module*` in comments.

Creastery committed:
Logic/commands: refactor `SelectCommand` class attributes

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `SelectCommand` class
by:
* refactoring class attribute names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
Model/util: refactor `SampleDataUtil#getSamplePersons()`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`SampleDataUtil#getSamplePersons()` to `getSampleModules()` and cascade
the changes to all affected classes.

Creastery committed:
Model: refactor `AddressBook` class methods

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `AddressBook` class
by:
* refactoring class attribute names containing `*person*` to `*module*`
* replacing `*person*` with `*module*` in comments
* cascading the changes to all affected classes

Creastery committed:
Model: refactor `ModelManager` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `ModelManager` class
by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring class attributes names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`
* replacing `*person*` with `*module*` in comments

Creastery committed:
Storage: refactor class `JsonAdaptedPerson` to `JsonAdaptedModule`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class
`JsonAdaptedPerson` to `JsonAdaptedModule` and:
* refactor variable names containing `*person*` to `*module*`
* cascade the changes to all affected classes

Creastery committed:
Storage: refactor `JsonSerializableAddressBook` class attributes

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`JsonSerializableAddressBook` class by:
* refactoring class attribute names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Note: List<JsonAdaptedModule> `persons` remains unchanged in this commit
as it will affect how Jackson serializes the AddressBook into JSON
format, causing unit tests to fail. As such, the class attribute
`persons` will only be refactored in the second phase of refactoring.

Creastery committed:
UI: refactor `BrowserPanel` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `BrowserPanel` class
by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`

Creastery committed:
UI: refactor class `PersonCard` to `ModuleCard`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class `PersonCard` to
`ModuleCard` and cascade the changes to all affected classes.

Creastery committed:
UI: refactor class `PersonListPanel` to `ModuleListPanel`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class
`PersonListPanel` to `ModuleListPanel` and:
* replace `*person*` with `*module*` in comments
* cascade the changes to all affected classes

Note: ListView<Module> `personListView` remains unchanged in this commit
as it will affect how JavaFX renders the FXML, which might cause unit
tests to fail. As such, this class attribute `personListView` will only
be refactored in the second phase of refactoring.

Creastery committed:
guitests: refactor class `PersonCardHandle` to `ModuleCardHandle`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class
`PersonCardHandle` to `ModuleCardHandle` and:
* replace `*person*` with `*module*` in comments
* cascade the changes to all affected classes

Creastery committed:
guitests: refactor `PersonListPanelHandle` to `ModuleListPanelHandle`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`PersonListPanelHandle` class to `ModuleListPanelHandle` and:
* refactoring class method names containing `*person*` to `*module*`
* refactor class attributes names containing `*person*` to `*module*`
* replacing `*person*` with `*module*` in comments
* cascading the changes to all affected classes

Creastery committed:
test/commons: refactor `IndexTest` class variable names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `fifthPersonIndex` to
`fifthModuleIndex`.

Creastery committed:
test/testutil: refactor `AddressBookBuilder#withPerson(...)`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`AddressBookBuilder#withPerson(..)` to `withModule(...)` and:
* replace `*person*` with `*module*` in comments
* cascade the changes to all affected classes

Creastery committed:
test/testutil: refactor class `PersonBuilder` to `ModuleBuilder`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class `PersonBuilder`
to `ModuleBuilder` and cascade the changes to all affected classes.

Creastery committed:
test/testutil: refactor class `PersonUtil` to `ModuleUtil`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class `PersonUtil`
to `ModuleUtil` and cascade the changes to all affected classes.

Creastery committed:
test/testutil: refactor `TestUtil#getPerson(...)` to `getModule(...)`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class
`TestUtil#getPerson(...)` to `getModule(...)` and cascade the changes to
all affected classes.

Creastery committed:
test/testutil: refactor `TypicalIndexes` class attributes

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `TypicalIndexes`
class by:
* refactoring class attribute names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
test/testutil: refactor class `TypicalPersons` to `TypicalModules`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor class
`TypicalPersons` to `TypicalModules` and cascade the changes to all
affected classes.

Creastery committed:
test/testutil: refactor `EditPersonDescriptorBuilder` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`EditPersonDescriptorBuilder` class to `EditModuleDescriptorBuilder` and
cascade the changes to all affected classes.

Creastery committed:
test/commands: refactor `AddCommand` and `AddCommandIntegrationTest`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `AddCommand` and
`AddCommandIntegrationTest` classes by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring class attributes names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`

Creastery committed:
test/commands: refactor `CommandTestUtil` class methods

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `CommandTestUtil`
class by:
* refactoring class method names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
test/commands: refactor `DeleteCommandTest` class methods

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `DeleteCommandTest`
class by:
* refactoring class method names containing `*person*` to `*module*`
* replacing `*person*` with `*module*` in comments

Creastery committed:
test/commands: refactor `EditCommandTest` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `EditCommandTest`
class by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`
* replacing `*person*` with `*module*` in comments

Creastery committed:
test/commands: refactor `FindCommandTest` class methods

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `FindCommandTest`
class by refactoring class method names containing `*person*` to
`*module*`.

Creastery committed:
test/commands: refactor `SelectCommandTest` class variable names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `lastPersonIndex` to
`lastModuleIndex`.

Creastery committed:
test/logic: refactor `LogicManagerTest` class methods

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `LogicManagerTest`
class by refactoring class method names containing `*person*` to
`*module*`

Creastery committed:
test/model: refactor class `PersonTest` to `ModuleTest`

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `PersonTest` class to
`ModuleTest` by refactoring class method names containing `*person*` to
`*module*`.

Creastery committed:
test/model: refactor `LogicManagerListTest` class methods names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`UniqueModuleListTest` class by refactoring class method names containing
`*person*` to `*module*`.

Creastery committed:
test/model: refactor `AddressBookTest` class method names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `AddressBookTest`
class and refactor class method names containing `*person*` to
`*module*`.

Creastery committed:
test/model: refactor `ModelManagerTest` class method names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `ModelManagerTest`
class and refactor class method names containing `*person*` to
`*module*`.

Creastery committed:
test/storage: refactor `JsonAdaptedModuleTest` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`JsonAdaptedModuleTest` class by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`

Creastery committed:
test/storage: refactor `JsonAddressBookStorageTest` class method names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`JsonAddressBookStorageTest` by refactoring method names containing
`*person*` to `*module*`.

Creastery committed:
test/storage: refactor `JsonAddressBookStorageTest` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`JsonAddressBookStorageTest` class by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring class attributes names containing `*person*` to `*module*`

Creastery committed:
test/ui: refactor `GuiTestAssert` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `GuiTestAssert` class
by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`
* cascading the changes to all affected classes

Creastery committed:
test/ui: refactor `BrowserPanelTest` class attributes

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`BrowserPanelTest` by refactoring class attribute names containing
`*person*` to `*module*`.

Creastery committed:
test/ui: refactor `ModuleListPanelTest` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`ModuleListPanelTest` class by:
* refactoring class attributes names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`
* replacing `*person*` with `*module*` in comments

Creastery committed:
systemtests: refactor `AddressBookSystemTest` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`AddressBookSystemTest` class by:
* refactoring class method names containing `*person*` to `*module*`
* replacing `*person*` with `*module*` in comments
* cascading the changes to all affected classes

Creastery committed:
systemtests: refactor `DeleteCommandSystemTest` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`DeleteCommandSystemTest` class by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`

Creastery committed:
systemtests: refactor `FindCommandSystemTest` class comments

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`FindCommandSystemTest` class by replacing `*person*` with `*module*` in
comments.

Creastery committed:
systemtests: refactor `ModelHelper` class

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor `ModelHelper` class
by:
* refactoring class method names containing `*person*` to `*module*`
* refactoring variable names containing `*person*` to `*module*`

Creastery committed:
systemtests: refactor `SelectCommandSystemTest` class variable names

We are in the first refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this first phase of refactoring, let's refactor
`SelectCommandSystemTest` class by refactoring variable names containing
`*person*` to `*module*`.

Creastery committed:
UI: refactor `ModuleListPanel#moduleListView` class attribute name

We are in the second refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this second phase of refactoring, let's refactor
`ModuleListPanel#personListView` to `moduleListView` and cascade the
changes to all affected classes.

Creastery committed:
UI: refactor `MainWindow#personListPanelPlaceholder` class attribute

We are in the second refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this second phase of refactoring, let's refactor
`MainWindow#personListPanelPlaceholder` to `moduleListPanelPlaceholder`
and cascade the changes to all affected classes.

Creastery committed:
UI: refactor `PersonListCard.fxml` to `ModuleListCard.fxml`

We are in the second refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this second phase of refactoring, let's refactor
`PersonListCard.fxml` to `ModuleListCard.fxml` and cascade the changes
to all affected classes.

Creastery committed:
UI: refactor `PersonListPanel.fxml` to `ModuleListPanel.fxml`

We are in the second refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this second phase of refactoring, let's refactor
`PersonListPanel.fxml` to `ModuleListPanel.fxml` and cascade the changes
to all affected classes.

Creastery committed:
Storage: refactor `JsonSerializableAddressBook#persons` to `modules`

We are in the second refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this second phase of refactoring, let's refactor
`JsonSerializableAddressBook#persons` class attribute to `modules` and
update all test files in `JsonAddressBookStorageTest` directory to
reflect the change in naming of the Jackson JsonProperty.

Creastery committed:
test/data: refactor `JsonAddressBookStorageTest` JSON files

We are in the second refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this second phase of refactoring, let's refactor the JSON test data
files for `JsonAddressBookStorageTest` by:
* refactoring JSON files to rename as `*module*` instead of `*person*`
* replacing `*person*` with `*module*` in the file contents

Creastery committed:
test/data: refactor `JsonSerializableAddressBookTest` JSON files

We are in the second refactoring phase of morphing `persons`-oriented
Address Book application to a `modules`-oriented Degree Planner
application.

In this second phase of refactoring, let's refactor the JSON test data
files for `JsonSerializableAddressBookTest` by:
* refactoring JSON files to rename as `*module*` instead of `*person*`
* replacing `*person*` with `*module*` in the file contents

Creastery committed:
Replace `*person*` with `*module*` in all string literals

We are almost done with the second refactoring phase of morphing
`persons`-oriented Address Book application to a `modules`-oriented
Degree Planner application.

Let's refactor all the string literals to replace `*person*` with
`*module*` to complete the second refactoring phase.

Creastery committed:
DeveloperGuide: rename filepaths of sequence diagrams

We are in the third and final refactoring phase of morphing
`persons`-oriented Address Book application to a `modules`-oriented
Degree Planner application.

Let's begin the third phase of refactoring by renaming the filepaths of
sequence diagrams referenced by the Developer Guide.

Creastery committed:
docs: update diagrams and images

We are in the third and final refactoring phase of morphing
`persons`-oriented Address Book application to a `modules`-oriented
Degree Planner application.

In this third phase of refactoring, let's update all the diagrams and
images shown in the Developer Guide to replace all `*person*` with
`*module*`.

Creastery committed:
docs: replace `*person*` with `*module*` in User and Developer guides

We are almost done with the third and final refactoring phase of
morphing `persons`-oriented Address Book application to a
`modules`-oriented Degree Planner application.

As a final step of this third refactoring phase, let's update the User
images shown in the Developer Guide to replace all instances of
`*person*` with `*module*`.

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void add_duplicateModule_throwsDuplicateModuleException() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void remove_nullModule_throwsNullPointerException() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/logic/LogicManagerTest.java:
```diff
@@ -88,19 +87,19 @@ public void execute_storageThrowsIoException_throwsCommandException() throws Exc
 
     @Test
-    public void getFilteredPersonList_modifyList_throwsUnsupportedOperationException() {
+    public void getFilteredModuleList_modifyList_throwsUnsupportedOperationException() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void setModule_editedModuleHasSameIdentity_success() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/storage/JsonAdaptedModuleTest.java:
```diff
@@ -33,78 +33,78 @@
-    public void toModelType_validPersonDetails_returnsPerson() throws Exception {
-        JsonAdaptedPerson person = new JsonAdaptedPerson(BENSON);
-        assertEquals(BENSON, person.toModelType());
+    public void toModelType_validModuleDetails_returnsModule() throws Exception {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void contains_moduleWithSameIdentityFieldsInList_returnsTrue() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void add_nullModule_throwsNullPointerException() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void setModule_nullEditedModule_throwsNullPointerException() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void setModule_targetModuleNotInList_throwsModuleNotFoundException() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void setModules_uniqueModuleList_replacesOwnListWithProvidedUniqueModuleList() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void setModules_nullUniqueModuleList_throwsNullPointerException() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void setModule_editedModuleHasDifferentIdentity_success() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void setModule_nullTargetModule_throwsNullPointerException() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void setModule_editedModuleIsSameModule_success() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void remove_existingModule_removesModule() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/model/module/UniqueModuleListTest.java:
```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void asUnmodifiableObservableList_modifyList_throwsUnsupportedOperationException() {
```
Resolving this issue as this is permitted according to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/index.html) Naming section:

> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()
> 
> e.g. sortList_emptyList_exceptionThrown() getMember_memberNotFount_nullReturned

Creastery commented on src/test/java/seedu/address/ui/ModuleCardTest.java:
```diff
@@ -0,0 +1,72 @@
+        assertTrue(moduleCard.equals(moduleCard));
+
+        // null -> returns false
+        assertFalse(moduleCard.equals(null));
```
`ModuleCardTest.java` is refactored from `PersonCardTest.java`, so no code has changed apart from the naming.

I'll open a new issue to look into heeding @codacy-bot's advice on using `==` to check for null instead of using `.equals()` as doing the latter can throw `NullPointerException` if `moduleCard` is null.

As the purpose of this PR is not to improve code quality but rather simply refactor `*person*` into `*module*`, I suggest we leave improving code quality for another PR and as such, I will resolve this.

lycjackie reviewed (changes requested):
[67/67]

There are still `people` references in the documentation in the developer guide

1. ... to show the total number of `people` in the address book... Line 640
2.  ... for the total number of `people` status, just like what we did for save location and sync status... Line 657

Creastery commented:
> [67/67]
> 
> There are still `people` references in the documentation in the developer guide
> 
> 1. ... to show the total number of `people` in the address book... Line 640
> 2. ... for the total number of `people` status, just like what we did for save location and sync status... Line 657

Thanks for spotting this. Will rectify immediately.

Creastery committed:
docs/DeveloperGuide: replace `*person*` with `*module*`

Previously, we replaced all `*person*` with `*module*` in the
`DeveloperGuide`. However, we left out `*people*`, which should be
replaced with `*modules*`.

Let's update the `DeveloperGuide` to replace `*people*` with
`*modules*`.

Creastery commented:
@lycjackie I've fixed the issue you mentioned in the latest commit [b442931]. 
Please re-review. Thanks!

Letm3through reviewed (approved):
LGTM :+1: . Thanks for the amazing work!

lycjackie commented:
LGTM 💯 

Thanks for all the hard work!

Creastery commented:
Thanks for the quick and detailed review! :)
